core/types/suave_structs.go

```diff
@@ -0,0 +1,32 @@
+// Code generated by suave/gen. DO NOT EDIT.
```

Comment:
Maybe it would make sense to separate the ABI generation from the rest of the changes (Impl wrapper)?

Reply:
Yeah a separate PR
I see two distinct changes - the autogeneration and the changes to contracts' Run functions (and SuavePrecompiledContractWrapper)